### PR TITLE
PyTest styled refactoring of tests.

### PR DESCRIPTION
**Pull request by quantshah**
- Rewrote `test_ops.py` using pytest.
- Removed loops and used `pytest.mark.parameterize`

**Benefits:**
#99 

I noticed that the tests for `cv.CrossKerr`raises a ValueError error and the other two (cv.CrossKerr and cv.CubicPhase) raise RuntimeError for the same test. I did not understand what is the proper exception to check so I split the tests. @josh146

### CODE REVIEW

**Comment by josh146** [comment]
Thanks @quantshah, looks great. I'll start code review tomorrow most likely 👍

**Comment by josh146** [comment]
With regards to the exceptions, `ValueError` seems odd in this case... my feeling is that all three should probably be `RuntimeError`.

**Comment by quantshah** [comment]
I will check the error part in a different PR. Thanks.

File: tests/test_ops.py [hunk 1]
```diff
@@ -1,173 +1,200 @@
...
+    def test_heisenberg_transformation_nongaussian(self):
+        """ops: Tests that proper exceptions are raised if we try to call the
+        Heisenberg transformation of non-Gaussian gates."""
+        op = cv.Kerr
+        with pytest.raises(RuntimeError, match=r"not a Gaussian operation"):
+            op_ = op(*[0.1] * op.num_params, [0] * op.num_wires, do_queue=False)
+            op_.heisenberg_tr(op.num_wires)
+
+        op = cv.CrossKerr
+        with pytest.raises(ValueError):
```
Comment (josh146) [inline tests/test_ops.py]: Just a reminder for us to make an issue on this `ValueError`, seems confusing to not have this match the others.

**Comment by josh146** [comment]
Looks great, merging this now

**Pull request merged**